py3.11 forbids inline flags such as (?i) after the first byte
pieced together from commits such as
https://source.chromium.org/chromium/chromium/src/+/f90f49df8db04dcb72f7ce0c4d0b2fe329bab00c
--- a/src/3rdparty/chromium/tools/metrics/structured/model.py
+++ b/src/3rdparty/chromium/tools/metrics/structured/model.py
@@ -26,7 +26,7 @@
     'metric',
     attributes=[
       ('name', unicode, r'^[A-Za-z0-9_.]+$'),
-      ('kind', unicode, r'^(?i)(|hashed-string|int)$'),
+      ('kind', unicode, r'(?i)^(|hashed-string|int)$'),
     ],
     alphabetization=[
         (_OBSOLETE_TYPE.tag, lambda _: 1),
--- a/src/3rdparty/chromium/tools/metrics/ukm/ukm_model.py
+++ b/src/3rdparty/chromium/tools/metrics/ukm/ukm_model.py
@@ -42,7 +42,7 @@
 _STATISTICS_TYPE =  models.ObjectNodeType(
     'statistics',
     attributes=[
-      ('export', str, r'^(?i)(|true|false)$'),
+      ('export', str, r'(?i)^(|true|false)$'),
     ],
     children=[
         models.ChildType(_QUANTILES_TYPE.tag, _QUANTILES_TYPE, multiple=False),
@@ -94,7 +94,7 @@
     'event',
     attributes=[
       ('name', str, r'^[A-Za-z0-9.]+$'),
-      ('singular', str, r'^(?i)(|true|false)$'),
+      ('singular', str, r'(?i)^(|true|false)$'),
     ],
     alphabetization=[
         (_OBSOLETE_TYPE.tag, _KEEP_ORDER),
